### DISCUSSION TITLE
how to enable the Ports tab beside terminal tab on UI

### DISCUSSION THREAD

**Discussion by vikaskonaparthi** [coder/code-server#5963] [Q&A]
How can I enable the Ports tab in the Code-Server web UI when using a Docker Compose file to create the container?

I am using this docker compose file 

`version: '2.2'
services:
  codeserver:
    container_name: "${containerName}"
    image: "${codeserverImage}"
    environment:
      PASSWORD: "${password}"
    cpus: "${CPUS_USG}"
    mem_limit: "${MEMORY_LIM}"
    
    ports:
      - "${port}:${containerPort}"
    volumes:
      - "${volume}:${containerVolume}"`
    
    I dont see ports tab here
    
    
![image](https://user-images.githubusercontent.com/44503709/212030878-41cefdb0-f085-440d-88a8-481fe30f7717.png)

**Comment by lg-nineoneninethreenine** [comment]
What? I can see one when I run my code-server. 

![image](https://user-images.githubusercontent.com/76845820/212068436-4c693bc7-b1b7-4bba-b74e-987821c1f06a.png)

What version of code-server do you use?

**Reply by vikaskonaparthi** [reply]
I am using docker to get the container up

**Comment by icarusion** [comment]
u can update to 4.9.1

**Comment by lg-nineoneninethreenine** [comment]
(this is a sort of continuation since https://github.com/coder/code-server/discussions/5963#discussioncomment-4667003)

It looks like you use Docker here, can you please share us your Dockerfile code? I think it is related to the update part or the tag you use.

**Comment by vikaskonaparthi** [comment]
This is the docker file 

`version: '2.2'
services:
codeserver:
container_name: codercom/code-server
image: "${codeserverImage}"
environment:
PASSWORD: "${password}"
cpus: 0.09
mem_limit: 300m
ports:
  - "${port}:${containerPort}"
volumes:
  - "${volume}:${containerVolume}"`

**Comment by vikaskonaparthi** [comment]
It got resolved I have updated it to 4.9.1 

but one more issue is 

![image](https://user-images.githubusercontent.com/44503709/212255895-e3ae26cf-190b-4875-9d49-967b994703ef.png)

When I run python flask application it is showing running on 5000 but when I check the running processes on my machine, I dont see any process running on 5000 port.

**Reply by jsjoeio** [reply]
Hmm so it must be running on 5000 inside the docker environment? I guess you'd need to expose that externally to access, right? Maybe @code-asher knows

**Reply by code-asher** [reply]
Yeah that sounds like a possibility.  What are you using to check the running processes and where/how are you running that command?

**Reply by vikaskonaparthi** [reply]
I am running this docker container on my linux machine and I am using ps -ef|grep flask to check the running process on the machine
and I am checking on my machine is anything running on port 5000 and dont see anything running up on the specific port.

**Reply by code-asher** [reply]
If you are running these commands in the host machine then I think the application will not show as running on 5000 since it is inside the Docker container.  It should show up when running inside the container though.
